src/types.c

```diff
@@ -133,6 +136,11 @@ bool is_number_type(const Type *t)
 {
     return is_integer_type(t) || t->kind == TYPE_DOUBLE;
 }
+
+bool is_float_type(const Type *t)
+{
+    return is_integer_type(t) || t->kind == TYPE_FLOAT;

```

Comment:
Integers shouldn't be considered floats. I see this function as something that checks whether you have `float` and `double`, similar to how `is_integer_type()` checks for `byte`/`int`/`long`.
```suggestion
    return t->kind == TYPE_FLOAT || t->kind == TYPE_DOUBLE;
```